app/src/main/java/io/gnosis/safe/ui/transaction/TransactionsFragment.kt

```diff
@@ -46,20 +48,38 @@ class TransactionsFragment : BaseViewBindingFragment<FragmentTransactionsBinding
             state.viewAction.let { viewAction ->
                 when (viewAction) {
                     is LoadTransactions -> loadTransactions(viewAction.newTransactions)
+                    is NoSafeSelected -> loadNoSafeFragment()
                     is BaseStateViewModel.ViewAction.ShowEmptyState -> showEmptyState()
+                    is ActiveSafeChanged -> handleActiveSafe(viewAction.activeSafe)
                     else -> binding.progress.visible(state.isLoading)
                 }
             }
         })
         viewModel.load()
     }
 
+    private fun loadNoSafeFragment() {
+        with(binding) {
+            transactions.visible(false)
+            progress.visible(false)
+            imageEmpty.visible(false)
+            labelEmpty.visible(false)
+            noSafe.apply {
+                childFragmentManager.beginTransaction()
+                    .add(noSafe.id, NoSafeFragment.newInstance(NoSafeFragment.Position.TRANSACTIONS))
```

Comment:
Probably replace instead of add would be better. Also here we might add fragment multiple times here. E.g. we have no safes, start add flow from transactions tab and after we done delete the newly added safe. In this case another fragment will be created and added. We should either reuse existing fragment or manually pop childFragmentManager's backstack when safe is loaded.

Reply:
good call. Should I just call `replace` instead of `add`?
```suggestion
                    .replace(noSafe.id, NoSafeFragment.newInstance(NoSafeFragment.Position.TRANSACTIONS))
```

Reply:
This will solve redundancy, but it will still hang around after safe is loaded, only in an invisible state. I think better to pop it off the stack